src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs

```diff
@@ -58,6 +58,15 @@ public async Task<HttpResponseMessage> ExecuteAsync(int retryCount = 15)
                         return response;
                     }
                 }
+                catch (HttpRequestException reqEx) when (reqEx.Message.Contains("404 (Not Found)"))
```

Comment:
Should we check the response.StatusCode for 404 before calling EnsureSuccessStatusCode() instead of looking at the exception message?